app/data_providers/dispatch.py

```diff
@@ -121,11 +121,36 @@ async def get_slate_lineup(
         ]
 
         return CorpusSlateLineupModel(
-            slates=flatten(list(
-                await gather(*slates)
-            )),
+            slates=self._dedupe_and_limit(
+                flatten(list(
+                    await gather(*slates)
+                )),
+                recommendation_count=recommendation_count,
+            ),
         )
 
+    def _dedupe_and_limit(
```

Comment:
Major int: You dont use self in this method. This method would have much less context if it were a classmethod.